docs/compass/03-01-security.md

```diff
@@ -16,30 +22,46 @@ The following diagram represents the architecture of the security in Compass:
-- `id`, which is the `authorization_id`
-- `tenant` (optional field - used for Application and Runtime, not used for Integration System)
-- `app_id` foreign key of type UUID
-- `runtime_id` foreign key of type UUID
-- `integration_system_id` foreign key of type UUID
-- `value` of type JSON, with authentication details, such as `client_id/client_secret` in the OAuth 2.0 authentication flow, or `username/password` in the Basic authentication flow. In the case of the certificates flow it is empty.
+- `id` - This is the `authorization_id`.
+- `tenant` - An optional field, used for Application and Runtime, and not used for Integration System.
+- `app_id` - A foreign key of type UUID
+- `runtime_id` - A foreign key of type UUID
+- `integration_system_id` - A foreign key of type UUID
+- `value` - A field of type JSON, with authentication details, such as, a `client_id`-`client_secret` in the OAuth 2.0 authentication flow, a Common Name in case of the certificates flow, or a token in the one-time-token flow.
+
+#### Custom Authenticator
+You can configure a custom JWT-based authentication for Compass with trusted issuers and locations for a tenant and scopes in the token. The flow is as follows:
+1. The authentication-mapping handler hydrator first checks for trusted issuer, next enriches, and then, provides the tenant-mapping handler with locations for tenant, scopes, and clientID.
+1. The tenant-mapping handler extracts them from the token's claims and verifies that the tenant exists.
+
+#### Consumer-Provider Flows
+As mentioned above, there are use cases where you can manage Compass resources from a different multi-tenant system, which shares the same tenancy model and knows the external tenant IDs, which are also used by Compass. In this case, Compass trusts those systems and allows them to manage resources on behalf of the user. The following options are valid for this flow:
+- The multi-tenant system is represented as an Integration System. In this case, the provider tenant is specified in the `Tenant` header. The scopes which are granted to the request, are the ones assigned to integration systems, by default. They are granted by the context provider, mentioned above.
+- The multi-tenant system is represented as a Runtime. In this case, the second context provider that matches is a custom authenticator, where the provider tenant is part of a JWT, along with the scopes that are granted. The scopes are taken from the context provider in this case.
+
+Both cases feature two context providers, which are used in a pair. One of context providers must be an externally-issued certificate context provider. It extracts the consumer tenant ID from the certificate. Later on, Compass checks if the consumer tenant has access to the provider tenant. For more information, see the [Authentication Flows](03-01-security.md#authentication-flows) section in this document.
```

Comment:
In the whole doc consumer and provider seems to be switched based on compass perspective. 

provider - the tenant providing system metadata (like ORD provider)
consumer - the tenant consuming system metadata (like ORD consumer)

However, the most common use of those terms in the SaaS registry world are from the SaaS app's perspective:

provider - the SaaS application that provides some features to a client
consumer - the client that consumes those features

Maybe we can stick to this terminology as most widely used in other resources.